frappe/permissions.py

```diff
@@ -73,6 +73,13 @@ def has_permission(doctype, ptype="read", doc=None, verbose=False, user=None, ra
 
 		role_permissions = get_role_permissions(meta, user=user)
 		perm = role_permissions.get(ptype)
+
+		if not perm and ptype == 'read':
+			# if not have perm for `read`
```

Comment:
why this? ptype is passed as a param, right? the function should not assume this.

Reply:
fixed with https://github.com/frappe/frappe/pull/12063/commits/a002b046ad33aa331f40db58f007c7e153ac1541